servers/physics_2d/body_2d_sw.h

```diff
@@ -129,6 +130,7 @@ class Body2DSW : public CollisionObject2DSW {
 	_FORCE_INLINE_ void _compute_area_gravity_and_dampenings(const Area2DSW *p_area);
 
 	friend class PhysicsDirectBodyState2DSW; // i give up, too many functions to expose
+	friend class BodyPair2DSW;
 
```

Comment:
`friend` should not be used. `public` and `private` variables are there for a reason. And, no, It shouldn't have been used in the line above either.

Reply:
Oh shoot, this was a remnant of some experimentation. It's not actually needed for the pointer implementation.